src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -990,6 +1004,12 @@ static inline intptr_t get_next_hash(Thread* current, oop obj) {
     v = (v ^ (v >> 19)) ^ (t ^ (t >> 8));
     current->_hashStateW = v;
     value = v;
+#ifdef ASSERT
+    // Verify that during CDS dumping, only a single thread
+    // ever calls ihash
+    assert(!CDSConfig::is_dumping_archive() || runs_on_one_thread_only(),
+        "Only one thread should generate ihash during CDS dumps");
```

Comment:
It should check for `CDSConfig::is_dumping_static_archive()` instead (here and in `Thread::Thread()`), to match the check in [JVM_StartThread]()https://github.com/openjdk/jdk/blob/7c1fad4fb6c387bbfb72b3f96b610e7cbc2ef312/src/hotspot/share/prims/jvm.cpp#L2946-L2948)